PowerPointLabs/PowerPointLabs/PositionsLab/PositionsPaneWPF.xaml.cs

```diff
@@ -1666,6 +1674,39 @@ private void SaveSelectedShapePositions(PowerPoint.ShapeRange shapes, Dictionary
                 dictionary.Add(s.Id, new PositionShapeProperties(new System.Drawing.PointF(s.Left, s.Top), s.Rotation, s.HorizontalFlip, s.VerticalFlip));
             }
         }
+
+        private void UpdateZOrder(Shape original, int zOrder)
```

Comment:
this ZOrder methods look generic, and can be used by other features.
try to put it into the util class